Enabling Syft seems to fail

When uncommenting the standard build, Syft appears to fail - unable to access Podman to operate on the container that has been built in the pipeline so far.

### Steps to reproduce
I've documented a minimal example in a fresh repo here that I think means this problem is going to be fairly reproducible:

* Fork repo from template as described in README.md: https://github.com/alexhaydock/testfin/commit/d35d47e19e3ff1fe6dd181511c40074501bb5d6a
* Update image name across files as described in README.md: https://github.com/alexhaydock/testfin/commit/446a32c9f23da942b810c782dca8d71fdcbdccab
* Enable SBOM by uncommenting Syft action: https://github.com/alexhaydock/testfin/commit/88df32d6d7d7d99dae8f719ab0a3ec6d76cae226
* Experience failure in Actions run: https://github.com/alexhaydock/testfin/actions/runs/25934785265/job/76238824562

### Failure mode
At the "Generate SBOM" step, the process fails with this error:

```
[0000] ERROR could not determine source: errors occurred attempting to resolve 'testfin:stable':
  - docker: pull failed: Error response from daemon: pull access denied for testfin, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
  - podman: podman not available: no host address
  - containerd: containerd not available: failed to dial "/run/containerd/containerd.sock": connection error: desc = "transport: error while dialing: dial unix /run/containerd/containerd.sock: connect: permission denied"
  - oci-registry: failed to get image descriptor from registry: GET https://index.docker.io/v2/library/testfin/manifests/stable: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:library/testfin Type:repository]]
  - additionally, the following providers failed with file does not exist: docker-archive, oci-archive, oci-dir, singularity, oci-dir, local-file, local-directory
Error: Process completed with exit code 1.
```

From what I can tell, there's something going on here that means that Syft can't find the image in Podman.

### Things I've tried so far (sadly none of which worked)
* Adding `systemctl start podman.service` into the Generate SBOM step, just in case it needed the Podman daemon/socket live and can't work with rootless.
* Bringing in the logic from the Bluefin upstream which seems much more complex/involved: https://github.com/ublue-os/bluefin/blob/9ba27cf682a2e90e0e897129c7095bb8010463c7/.github/workflows/reusable-build.yml#L145-L164
* Bringing in just the newer version of Syft used in the upstream `reusable-build.yml` file in case it's an issue fixed in newer Syft versions
* Tried enabling image signing with Cosign as README.md suggests the Syft step "Requires image signing to be enabled first" - that didn't help since I think this only applies to the "Add SBOM Attestation" step and not the earlier "Generate SBOM" step where the failure actually is

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -146,9 +146,12 @@ jobs:
       #     DEFAULT_TAG: ${{ env.DEFAULT_TAG }}
       #     SYFT_CMD: ${{ steps.setup-syft.outputs.cmd }}
       #   run: |
+      #     # Save image to tar so Syft can access it (podman rootless has no socket)
+      #     IMAGE_TAR="$(mktemp -d)/image.tar"
+      #     podman save -o "$IMAGE_TAR" ${IMAGE}:${DEFAULT_TAG}
       #     OUTPUT_PATH="$(mktemp -d)/sbom.json"
       #     export SYFT_PARALLELISM=$(($(nproc)*2))
-      #     $SYFT_CMD ${IMAGE}:${DEFAULT_TAG} -o spdx-json=${OUTPUT_PATH}
+      #     $SYFT_CMD docker-archive:$IMAGE_TAR -o spdx-json=${OUTPUT_PATH}
       #     echo "OUTPUT_PATH=${OUTPUT_PATH}" >> $GITHUB_OUTPUT
 
       # These `if` statements are so that pull requests for your custom images do not make it publish any packages under your name without you knowing

